R/survfit2.R

```diff
@@ -91,5 +91,29 @@ survfit2 <- function(formula, ...) {
   # update object with env and add another class -------------------------------
   survfit %>%
     utils::modifyList(val = list(.Environment = env)) %>%
-    structure(class = c("survfit2", class(survfit)))
+    structure(class = c("survfit2", class(survfit))) %>%
+    .check_PARAM_consistency()
+}
+
+.check_PARAM_consistency <- function(x) {
+  # get the formula and data
+  formula <- .extract_formula_from_survfit(x)
+  data <- .extract_data_from_survfit(x)
+
+  if (is.null(data) || is.null(formula)) return(x)
+  if (all(c("PARAM", "PARAMCD") %in% names(data)) && !.is_PARAM_consistent(formula, data))
+    cli::cli_inform(c("!" = "Columns {.cls {c('PARAM', 'PARAMCD')}} are not unique and usage is likely incorrect."))
```

Comment:
This should be an error and stop the estimation. Most clinical programmers don't check their logs and we might use the function inside a wrapper where this message might get lost otherwise.

Reply:
I upgraded it from a message to a warning. In the future a wrapper can upgrade the warning to an error if they prefer with something like `withr::with_options(new = list(warn = 2), <wrapper code here>)`.

I honestly don't want any excess code is `survfit2()` that makes it different from `survival::survfit()`, but I think this is a reasonable middle ground.